rotkehlchen/chain/ethereum/trades.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import annotations
+
+from typing import (
+    Any,
+    Dict,
+    NamedTuple,
+    Tuple,
+    Union,
+)
+
+from rotkehlchen.assets.asset import EthereumToken
+from rotkehlchen.assets.unknown_asset import (
+    FakeAsset,
+    UnknownEthereumToken,
+)
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_ethereum_address,
+    deserialize_ethereum_token_from_db,
+    deserialize_fee,
+    deserialize_location_from_db,
+    deserialize_price,
+    deserialize_timestamp,
+    deserialize_trade_type_from_db,
+    deserialize_unknown_ethereum_token_from_db,
+)
+from rotkehlchen.typing import (
+    AssetAmount,
+    ChecksumEthAddress,
+    Fee,
+    Location,
+    Price,
+    Timestamp,
+    TradeType,
+)
+
+# Quick lookup for AMMTrade locations
+AMMTradeLocations = {Location.UNISWAP}
+AMMTradeDBTuple = (
+    Tuple[
+        str,  # tx_hash
+        int,  # log_index
+        str,  # address
+        int,  # timestamp
+        str,  # location
+        str,  # type
+        int,  # is_base_asset_unknown
+        str,  # base_asset_address
+        str,  # base_asset_symbol
+        str,  # base_asset_name
+        int,  # base_asset_decimals
+        int,  # is_quote_asset_unknown
+        str,  # quote_asset_address
+        str,  # quote_asset_symbol
+        str,  # quote_asset_name
+        int,  # quote_asset_decimals
+        str,  # amount
+        str,  # rate
+        str,  # fee
+        str,  # notes
+    ]
+)
+
+
+class AMMTrade(NamedTuple):
+    """This class aims for a better trades support than the current Trade class
+    in AMMs protocols.
+    """
+    tx_hash: str
+    log_index: int
+    address: ChecksumEthAddress
+    timestamp: Timestamp
+    location: Location
+    trade_type: TradeType
+    base_asset: Union[EthereumToken, UnknownEthereumToken]
+    quote_asset: Union[EthereumToken, UnknownEthereumToken]
+    amount: AssetAmount
+    rate: Price
+    fee: Fee
+    notes: str = ''
+
+    @classmethod
```

Comment:
That is not needed. You can do it easier as a static method.

Example from my editor:

```python
    @staticmethod
    def deserialize_from_db(value: str) -> 'BalanceType':
        if value == 'A':
            return BalanceType.ASSET
        elif value == 'B':
            return BalanceType.LIABILITY

        raise DeserializationError(f'Encountered unknown BalanceType value {value} in the DB')
```

Reply:
For me it clearly is a `classmethod`, as it interacts with the class (in fact it works as a factory!). This topic is quite popular in the Python community, isn't it? All I've read before is that no one stands for static methods.

Reply:
class and static methods are almost the same thing.

"No one stands for static methods" is a silly argument.

For me it's a simple choice. For a class method you need to also pass the class object as the first argument. For a static method you don't. The end result is the same.

So less code, easier to read and understand. Staticmethod wins.

Reply:
Hmm what I am saying above is not exactly true.

Classmethod is indeed better for a factory method. No need to change this.
Reasons:

1.The argument is implicit. You don't pass it. So usage is the same as staticmethod
2. It works great as a factory **even** for subclasses since the proper sublass is passed. This would not work the same with a static method

On the other hand staticmethods are good when you want to group a function into the class for code organization purposes.

I am going to mark this comment as resolved.